Automattic/wp-calypso
# VideoPress Onboarding: don't show site preview for VideoPress flow if a supported plan has not been purchased

#### Proposed Changes

* prevent iframe from loading the site preview if a plan has not been purchased, since the preview won't have the VideoMaker theme applied
* add a new site option to allow D97066-code to better detect the purchase so the theme is properly applied after plan purchase outside of the onboarding flow

<img width="722" alt="Screenshot 2023-01-04 at 3 43 05 PM" src="https://user-images.githubusercontent.com/4081020/210646372-0691a4b5-37ca-4cf9-b49d-ca21f3ce2434.png">


#### Testing Instructions



* visit `/setup/videopress`
* Go through the flow up to the Checkout page, do not check out
* Press the `X` in the upper left to cancel
* Select either `Empty Cart` or `Leave items`
* You will be brought to the wpcom dashboard and will NOT see the site preview
* Press the `W` logo in the upper left
* You will be returned to the Launchpad
* Press Choose a Plan
* Select a plan
* Complete the purchase and skip the Plugins upgrade question (if you see it)
* Press the `W` logo again
* You should be returned to the Launchpad
* You will see the site preview, with the correct theme!

#### Pre-merge Checklist





Related to #
